### PR TITLE
test: add non swap multi call

### PR DESCRIPTION
**Pull request by 0xsambugs**
We now have four multi call tests that do not involve swaps:
- Withdraw swapped and create a new position with it (already existed, just refactored)
- Withdraw swapped and unswapped balance in one tx, without terminating (already existed, just refactored)
- Take from caller and create multiple positions
- Withdraw swapped and then distribute to many recipients

### CODE REVIEW

File: test/integration/DCAHubCompanion/multi-call.spec.ts [hunk 1]
```diff
@@ -46,22 +52,32 @@ contract('Multicall', () => {
     DCAHubCompanion = await ethers.getContract('DCAHubCompanion');
     DCAHubSwapper = await ethers.getContract('DCAHubSwapper');
     DCAPermissionManager = await ethers.getContract('PermissionsManager');
+    transformerRegistry = await ethers.getContract('TransformerRegistry');
 
-    await ethers.provider.send('hardhat_setBalance', [DCAHubSwapper.address, '0x0']);
-
-    // Allow one minute interval
-    await DCAHub.connect(admin).addSwapIntervalsToAllowedList([SwapInterval.ONE_MINUTE.seconds]);
+    // const swapperRegistry = await ethers.getContract<SwapperRegistry>('SwapperRegistry');
```
Comment (0xsambugs) [inline test/integration/DCAHubCompanion/multi-call.spec.ts:57]: Left this here for the next PR

File: test/integration/DCAHubCompanion/multi-call.spec.ts [hunk 2]
```diff
@@ -46,22 +52,32 @@ contract('Multicall', () => {
...
 
     // Allow tokens and swapper
-    await DCAHub.connect(admin).setAllowedTokens([WETH_ADDRESS, USDC_ADDRESS], [true, true]);
+    await DCAHub.connect(admin).setAllowedTokens([WETH_ADDRESS, USDC_ADDRESS, WBTC_ADDRESS], [true, true, true]);
     await DCAHubSwapper.connect(admin).grantRole(await DCAHubSwapper.SWAP_EXECUTION_ROLE(), swapper.address);
 
     // Send tokens from whales, to our users
     await distributeTokensToUsers();
 
+    // await swapperRegistry.connect(admin).allowSwappers([
```
Comment (0xsambugs) [inline test/integration/DCAHubCompanion/multi-call.spec.ts:72]: Same

File: test/integration/DCAHubCompanion/multi-call.spec.ts [hunk 3]
```diff
@@ -461,6 +578,61 @@ contract('Multicall', () => {
...
+
+  async function sendToRecipient({ token, amount, recipient }: { token: IERC20; amount: BigNumberish; recipient: HasAddress }) {
+    const { data } = await DCAHubCompanion.populateTransaction.sendToRecipient(token.address, amount, recipient.address);
+    return data!;
+  }
+
+  async function sendAllInCompanionToRecipient({ token, recipient }: { token: IERC20; recipient: HasAddress }) {
+    const { data } = await DCAHubCompanion.populateTransaction.sendBalanceOnContractToRecipient(token.address, recipient.address);
+    return data!;
+  }
```
Comment (0xged) [inline test/integration/DCAHubCompanion/multi-call.spec.ts:634]: This two don't have the `Data` suffix, I really don't mind it, but maybe for consistency?

**Review by 0xged — APPROVED** [review]
Looks pretty... f-ing good! AND IT'S WORKING! 🚀

File: test/integration/DCAHubCompanion/multi-call.spec.ts [hunk 4]
```diff
@@ -70,6 +86,135 @@ contract('Multicall', () => {
...
+        await DCAHubCompanion.multicall([permissionData, withdrawToCompanionData, sendHalfData, sendOtherHalfData]);
+      });
+      then(`hub's WETH balance is reduced`, async () => {
+        await expectBalanceToBe(WETH, DCAHub, hubToBalance.sub(swappedBalance));
+      });
+      then(`first recipients's WETH increases`, async () => {
+        await expectBalanceToBe(WETH, recipient, swappedBalance.div(2));
+      });
+      then(`second recipients's WETH increases`, async () => {
+        await expectBalanceToBe(WETH, otherRecipient, swappedBalance.sub(swappedBalance.div(2)));
```
[the remark this answers is not in the document]
Comment (0xsambugs) [reply test/integration/DCAHubCompanion/multi-call.spec.ts:212]: If `swappedBalance` is not divisible by two, then this change won't work

`11/2 = 5`. One recipient will get `5` and the other should get `6`

File: test/integration/DCAHubCompanion/multi-call.spec.ts (hunk 3 above)
Comment (0xsambugs) [reply test/integration/DCAHubCompanion/multi-call.spec.ts]: Added!

**Pull request merged**